### PR TITLE
Bug 1881542: Set all CO conditions on meeting expected sync ratio

### PR DESCRIPTION
**Pull request by anik120**
**Description of the change:**

On meeting the expected sync ratio of 0.3, the marketplace operator
was only setting the DEGRADED and PROGRESSING conditions. The
AVAILABLE and PROGRESSING conditions were not being set. This PR fixes
that issue.

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/status/status.go [hunk 1]
```diff
@@ -347,8 +347,12 @@ func (r *reporter) monitorClusterStatus() {
 				}
 				if isSucceeding {
 					statusConditions = conditionListBuilder(configv1.OperatorDegraded, configv1.ConditionFalse, fmt.Sprintf("Current CR sync ratio (%g) meets the expected success ratio (%g)", *ratio, successRatio), "OperandTransitionsSucceeding")
+					statusConditions = conditionListBuilder(configv1.OperatorAvailable, configv1.ConditionTrue, fmt.Sprintf("Available release version: %s", r.version), "OperandTransitionsSucceeding")
```
Comment (awgreene) [inline pkg/status/status.go:350]: Do we need to set statusConditions each time?

**Comment by awgreene** [comment]
Is [this code](https://github.com/operator-framework/operator-marketplace/blob/f5c44499cc171a8d70ba1a86b925d3cb135391ed/pkg/status/status.go#L323-L336) needed anymore?

File: pkg/status/status.go (hunk 1 above)
Comment (anik120) [reply pkg/status/status.go:350]: A good point. That can be cleaned up a bit.

**Comment by anik120** [comment]
> Is this code needed anymore?

@awgreene we have the default operatorsources in this release branch, and we do keep a count of the syncs. They were removed in the release-4.6 branch, and therefore the code too. We do want to preserve this logic for this branch.

**Comment by awgreene** [comment]
> > Is this code needed anymore?
> 
> @awgreene we have the default operatorsources in this release branch, and we do keep a count of the syncs. They were removed in the release-4.6 branch, and therefore the code too. We do want to preserve this logic for this branch.

@anik120 I was pointing out that we set all conditions at once with your change. It seems like we no longer need the code I linked.

**Comment by anik120** [comment]
@awgreene I see what you mean. Updated the PR.

File: pkg/status/status.go [hunk 2]
```diff
@@ -321,34 +321,22 @@ func (r *reporter) monitorClusterStatus() {
 			}
 
 			// Report that marketplace is available after meeting minimal syncs.
```
Comment (awgreene) [inline pkg/status/status.go:323]:
```suggestion
			// Report marketplace ClusterOperator conditions after meeting minimal syncs.
```

File: pkg/status/status.go [hunk 3]
```diff
@@ -321,34 +321,22 @@ func (r *reporter) monitorClusterStatus() {
...
 				conditionListBuilder := clusterStatusListBuilder()
-				if operatorUpgradeable {
-					conditionListBuilder(configv1.OperatorUpgradeable, configv1.ConditionTrue, upgradeableMessage, "OperatorAvailable")
+				conditionListBuilder(configv1.OperatorProgressing, configv1.ConditionFalse, fmt.Sprintf("Successfully progressed to release version: %s", r.version), "OperatorAvailable")
+				if isSucceeding {
+					conditionListBuilder(configv1.OperatorDegraded, configv1.ConditionFalse, fmt.Sprintf("Operator and Operands(OperatorSources) are available"), "OperandTransitionsSucceeding")
+					conditionListBuilder(configv1.OperatorAvailable, configv1.ConditionTrue, fmt.Sprintf("Available release version: %s", r.version), "OperatorAvailable")
 				} else {
-					conditionListBuilder(configv1.OperatorUpgradeable, configv1.ConditionFalse, deprecatedAPIMessage, "DeprecatedAPIsInUse")
+					conditionListBuilder(configv1.OperatorDegraded, configv1.ConditionTrue, fmt.Sprintf("One(1) or more OperatorSources are unavailable. Please inspect the openshift-marketplace namespace (oc get opsrc -n openshift-marketpalace) for more information"), "OperandTransitionsFailing")
```
Comment (awgreene) [inline pkg/status/status.go:333]:
```suggestion
					conditionListBuilder(configv1.OperatorDegraded, configv1.ConditionTrue, fmt.Sprintf("One or more OperatorSources are unavailable. Please inspect the openshift-marketplace namespace for more information"), "OperandTransitionsFailing")
```

**Review by awgreene — COMMENTED** [review]
Nice changes! Left a few nits.

**Comment by anik120** [comment]
@awgreene thanks for reviewing the PR. Merged your nits with this PR. PTAL.

**Pull request merged**